src/metabase/driver/mongo/query_processor.clj

```diff
@@ -390,14 +452,40 @@
        :collection source-table-name
        :mbql?      true})))
 
+;; omf: if value 'is' a ISODate, extract the date/time string and convert it to date object
+(defn- to-date-object [value]
+  (if (and (string? value) (clojure.string/starts-with? value "ISODate"))
+    (let [reg (re-seq #"ISODate\((.*?)\)" value)
+          isodate (get (first reg) 1)]
+          (c/from-long (c/to-long isodate)))
```

Comment:
You should use `re-find` and destructuring here. You shouldn't need to use `get` unless you have a really good reason for doing so (`ffirst` would have also worked in this situtation.)

e.g.

```clojure
(defn- to-date-object [value]
  (if-let [[_ date-string] (when (string? value)
                             (re-find #"^ISODate\((.*?)\)$" value))]
    (c/from-long (c/to-long date-string))
    value))
```